### PR TITLE
[REVIEW] Fix missing dtype handling in cudf.Series & columnops.as_column

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #2319

### CODE REVIEW

File: python/cudf/cudf/dataframe/series.py [hunk 1]
```diff
@@ -86,12 +86,16 @@ def __init__(
         elif isinstance(data, Index):
             name = data.name
             data = data.as_column()
+            if dtype is not None:
```
Comment (beckernick) [inline python/cudf/cudf/dataframe/series.py]: @galipremsagar if you've changed `as_column` to use the dtype argument on Indexes, could you use it that way here? If dtype=None, nothing will change. if dtype is not none, you'll typecast.

Comment (beckernick) [reply python/cudf/cudf/dataframe/series.py]: Chatted offline. We probably want this behavior to be:

1) Default to the existing dtype of an object, if it has one
2) Optionally override that default based on the keyword argument

@shwina may want to weigh in if he has a different thinking around type handling

File: python/cudf/cudf/tests/test_dataframe.py [hunk 2]
```diff
@@ -3288,6 +3288,32 @@ def test_as_column_types():
 
     assert_eq(pds, gds)
 
+    pds = pd.Series([])
```
Comment (shwina) [inline python/cudf/cudf/tests/test_dataframe.py]: Could we also add the specific test cases in #2319, i.e.?

```
print(columnops.as_column(cudf.Series([]), 
print(columnops.as_column(cudf.Series([]), dtype='float32').dtype)
print(columnops.as_column(cudf.Series([]), dtype='str').dtype)
```

**Comment by shwina** [comment]
Looks great! Just a comment about adding a couple more tests.

File: python/cudf/cudf/tests/test_dataframe.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_dataframe.py]: @shwina Added them into unit-tests 👍

**Pull request merged**